lib/spack/spack/test/cmd/location.py

```diff
@@ -65,7 +62,7 @@ def test_location_source_dir_missing():
     prefix = "==> Error: "
     expected = (
         "%sSource directory does not exist yet. Run this to create it:"
-        "%s  spack stage %s" % (prefix, os.linesep, spec)
+        "%s  spack stage %s" % (prefix, "\n", spec)
```

Comment:
I looked into this and I think that what you have currently should generally work because:

* `tty.die`/`tty.info` separate lines with `\n` (vs. os.linesep)
* `\n` works as a newline on latest Windows

However, the following introduces a potential for confusion:

* `\n` will be transformed to `\r\n` when written to text streams
* Opening in text mode always converts `\r\n` to `\n`, but opening in binary mode will preserve`\r\n`

While I expect to see `\n` for most combinations of writing/reading, it's possible to read `\r\n` (admittedly, only if `SpackCommand` logic is altered), and either is acceptable in this context.

That all being said, this was the only spot in our unit tests that checks for `linesep` vs. `\n`, and so I think in the context of this smaller PR, this change is the right way to go. Generally, it may be more robust to allow for either `\n` or `\r\n`.